pkg/client/client.go

```diff
@@ -180,77 +209,191 @@ func (o *OptimizelyClient) GetFeatureVariableBoolean(featureKey, variableKey str
 
-	val, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	val, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	var convertedValue int
+	defer func() {
+		if o.notificationCenter != nil {
+			variableMap := map[string]interface{}{
+				"variableKey":   variableKey,
+				"variableType":  valueType,
+				"variableValue": convertedValue,
+			}
+			if err != nil {
+				variableMap["variableValue"] = val
+			}
+			decisionNotification := decision.FeatureNotificationWithVariables(featureKey, featureDecision, &userContext, variableMap)
+			decisionNotification.Type = notification.FeatureVariable
+
+			if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+				o.logger.Warning("Problem with sending notification")
+			}
+		}
+	}()
 	if err != nil {
-		return 0, err
+		return convertedValue, err
 	}
-	convertedValue, err := strconv.Atoi(val)
+	convertedValue, err = strconv.Atoi(val)
 	if err != nil || valueType != entities.Integer {
 		return 0, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
 	}
+
 	return convertedValue, err
 }
 
 // GetFeatureVariableString returns the feature variable value of type string associated with the given feature and variable keys.
 func (o *OptimizelyClient) GetFeatureVariableString(featureKey, variableKey string, userContext entities.UserContext) (value string, err error) {
 
-	value, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	value, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+
+	defer func() {
+		if o.notificationCenter != nil {
+			variableMap := map[string]interface{}{
+				"variableKey":   variableKey,
+				"variableType":  valueType,
+				"variableValue": value,
+			}
+
+			decisionNotification := decision.FeatureNotificationWithVariables(featureKey, featureDecision, &userContext, variableMap)
+			decisionNotification.Type = notification.FeatureVariable
+
+			if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+				o.logger.Warning("Problem with sending notification")
+			}
+		}
+	}()
 	if err != nil {
 		return "", err
 	}
 	if valueType != entities.String {
 		return "", fmt.Errorf("variable value for key %s is wrong type", variableKey)
 	}
+
 	return value, err
 }
 
 // GetFeatureVariableJSON returns the feature variable value of type json associated with the given feature and variable keys.
 func (o *OptimizelyClient) GetFeatureVariableJSON(featureKey, variableKey string, userContext entities.UserContext) (value *optimizelyjson.OptimizelyJSON, err error) {
 
-	val, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	val, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	defer func() {
+		if o.notificationCenter != nil {
+			var variableValue interface{}
+			if value != nil {
```

Comment:
When would `value` be `nil`? I thought we only allowed JSON variables to be objects at the top level?

Reply:
OptimizelyJSON is the object and it can be nil, since I return the pointer to the object.  GetFeatureVariableJSON returns *optimizelyjson.OptimizelyJSON

Reply:
If it is `ni`l, is that an unexpected error condition, or does it happen expectedly in some case(s)?

Reply:
there is only one scenario where this becomes nil - when it cannot unmarshal properly. Since datafiles are providing all the variables as quoted strings, we need to unmarshal this first. If we cannot for some reason, then it's returning nil json object. So this is an expected error condition.